keel-echo/src/main/kotlin/com/netflix/spinnaker/keel/echo/ManualJudgementNotifier.kt

```diff
@@ -31,32 +32,41 @@ class ManualJudgementNotifier(
   private val echoService: EchoService,
   private val repository: KeelRepository,
   private val scmInfo: ScmInfo,
+  private val springEnv: Environment,
   @Value("\${spinnaker.baseUrl}") private val spinnakerBaseUrl: String
 ) {
   companion object {
     const val MANUAL_JUDGEMENT_DOC_URL =
       "https://www.spinnaker.io/guides/user/managed-delivery/environment-constraints/#manual-judgement"
   }
 
+  private val isNewSlackEnabled: Boolean
+    get() = springEnv.getProperty("keel.notifications.slack", Boolean::class.java, true)
+
   private val log by lazy { LoggerFactory.getLogger(javaClass) }
 
   @EventListener(ConstraintStateChanged::class)
   fun constraintStateChanged(event: ConstraintStateChanged) {
-    log.debug("Received constraint state changed event: $event")
-    // if this is the first time the constraint was evaluated, send a notification
-    // so the user can react via other interfaces outside the UI (e.g. e-mail, Slack)
-    if (event.constraint is ManualJudgementConstraint &&
-      event.previousState == null &&
-      event.currentState.status == ConstraintStatus.PENDING
-    ) {
-      event.environment.notifications.map {
-        // TODO: run in parallel
-        runBlocking {
-          log.debug("Sending notification for manual judgement with config $it")
-          echoService.sendNotification(event.toEchoNotification(it))
+    if (!isNewSlackEnabled) {
```

Comment:
Maybe a simpler way of doing this would be to make the entire `@Component` conditional on the config property, with a `@ConditionalOnProperty("keel.notifications.slack", havingValue = false, matchIfMissing = true)`.

Reply:
see my comment above - do you think i should change it if i plan to remove it in the next couple of days?

Reply:
Yeah, if you're planning to remove the condition altogether than it's probably not worth it.